vissl/data/data_helper.py

```diff
@@ -24,6 +24,65 @@ def get_mean_image(crop_size):
     return img
 
 
+@contextlib.contextmanager
+def with_temporary_numpy_seed(seed: int):
+    """
+    Context manager to run a specific portion of code with a given seed:
+    resumes the numpy state after the execution of the block
+    """
+    random_state = np.random.get_state()
+    np.random.seed(seed)
+    yield
+    np.random.set_state(random_state)
+
+
+def unbalanced_sub_sampling(
+    total_size: int, nb_samples: int, skip_samples: int = 0, seed: int = 0
+) -> np.ndarray:
+    """
+    Given an original dataset of size 'total_size', sub_sample part of the dataset such that
+    the sub sampling is deterministic (identical across distributed workers)
+    Return the selected indices
+    """
+    with with_temporary_numpy_seed(seed):
+        return np.random.choice(
+            total_size, size=skip_samples + nb_samples, replace=False
+        )[skip_samples:]
+
+
+def balanced_sub_sampling(
+    labels: np.ndarray, nb_samples: int, skip_samples: int = 0, seed: int = 0
+) -> np.ndarray:
+    """
+    Given some labels, sub_sample a part of the labels such that:
+    - the number of samples of each label differs by at most one
+    - the sub sampling is deterministic (identical across distributed workers)
+    Return the selected indices
+    """
+    groups = {}
+    for i, label in enumerate(labels):
```

Comment:
does this have any assumption on the labels type being list, numpy array etc ?

Reply:
The type hints says that the "labels" should be a numpy array, but this could be relaxed to any kind of Sequence. The other assumption is that the labels inside it should be hashable.

Reply:
docstring is perfect :) also , since labels can be many things, we can remove the type hint for labels or use the `Union` to compose the typehint :)